pilot/pkg/serviceregistry/serviceentry/conversion.go

```diff
@@ -191,19 +192,20 @@ func convertServices(cfg config.Config) []*model.Service {
 			hostAddresses = append(hostAddresses, &HostAddress{hostname, constants.UnspecifiedIP})
 		}
 	}
+	meshExternal := serviceEntry.Location == networking.ServiceEntry_MESH_EXTERNAL
 
-	return buildServices(hostAddresses, cfg.Namespace, svcPorts, serviceEntry.Location, resolution,
+	return buildServices(hostAddresses, cfg.Namespace, svcPorts, meshExternal, resolution,
 		exportTo, labelSelectors, serviceEntry.SubjectAltNames, creationTime, cfg.Labels)
 }
 
-func buildServices(hostAddresses []*HostAddress, namespace string, ports model.PortList, location networking.ServiceEntry_Location,
+func buildServices(hostAddresses []*HostAddress, namespace string, ports model.PortList, meshExternal bool,
```

Comment:
What is the purpose of this change? Passing location seems better so that `buildServices` can use it for other purpose if needed?

Reply:
> What is the purpose of this change? Passing location seems better so that `buildServices` can use it for other purpose if needed?

I think this function is just for constructing mode.Service[], MeshExternal should be the same as other parameters, and can be used after passing it, without needing to judge again. I haven't thought of other scenarios that Locality can use yet.

Reply:
I understand it is not used now but can be in future and I see no benefit of judging it before hand.